ocaml/message-switch/switch/switch_main.ml

```diff
@@ -75,6 +75,13 @@ module Lwt_result = struct
   let ( >>= ) m f = m >>= fun x -> f (Stdlib.Result.get_ok x)
 end
 
+let exn_hook e =
+  let bt = Printexc.get_raw_backtrace () in
+  error "Caught exception in Lwt.async: %s\n%s" (Printexc.to_string e)
```

Comment:
I think using `\n` in log messages is not a good idea. I believe it gets escaped anyway and so it would be better to use something else.

Reply:
maybe we have something in the debug module already to print a backtrace?